tests/javascript/unit/utils/textToTask.spec.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { textToTask } from '../../../../src/utils/textToTask.js'
+
+describe('utils/textToTask test suite', () => {
+	it('should convert simple list with empty lines', () => {
+		const text
+			= `    
+- task1
+	
+- task2
+- task3
+`
+
+		const tasks = textToTask(text)
+
+		expect(tasks).toEqual({
+			numberOfTasks: 3,
+			tasks: [
+				{ summary: 'task1', children: [] },
+				{ summary: 'task2', children: [] },
+				{ summary: 'task3', children: [] },
+			],
+		})
+	})
+
+	it('should convert different prefixes', () => {
+		const text
+			= `
+task
+- task
++ task
+* task
+- [] task
+- [x] task
+- [X] task
```

Comment:
I was under the impression, that this would create already completed tasks. While the markdown would suggest that, I am not sure, it would make sense.

Reply:
True, we could create completed tasks, but I'm not sure what's the use case for that?